src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,127 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWTH_FACTORY = 1.5;
+    private static final int INVALID_INDEX = -1;
+
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        this(DEFAULT_CAPACITY);
+    }
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity < 0) {
+            throw new IllegalStateException("Illegal capacity: " + initialCapacity);
+        }
+        this.elements = new Object[initialCapacity];
+        this.size = 0;
+    }
+
+    public void ensureCapacity(int minCapacity) {
```

Comment:
All methods used in the middle of the class must be private
```suggestion
    private void ensureCapacity(int minCapacity) {
```